task_manager/users/views.py

```diff
@@ -85,7 +86,15 @@ def get(self, request: HttpRequest, **kwargs: int | str) -> HttpResponse:
     def post(self, request: HttpRequest, **kwargs: int | str) -> HttpResponse:
         user_id = kwargs.get("id")
         user = get_object_or_404(ServiceUser, pk=user_id)
-        user.delete()
-        message = _("User has been deleted successfully")
-        messages.add_message(request, messages.SUCCESS, message)
+        user_author_tasks = Task.objects.filter(author=user)
+        user_executor_tasks = Task.objects.filter(executor=user)
+        if user_author_tasks or user_executor_tasks:
+            message = _(
+                "It is not possible to delete the user because it is being used"
+            )
+            messages.add_message(request, messages.ERROR, message)
+        else:
+            user.delete()
+            message = _("User has been deleted successfully")
+            messages.add_message(request, messages.SUCCESS, message)
```

Comment:
wdyt about this option?
```suggestion
        tasks_where_user_is_author = Task.objects.filter(author=user)
        tasks_where_user_is_executor = Task.objects.filter(executor=user)
        if tasks_where_user_is_author or tasks_where_user_is_executor:
            message = _(
                "It is not possible to delete the user because it is connected to tasks"
            )
            messages.add_message(request, messages.ERROR, message)
        else:
            user.delete()
            message = _("User has been deleted successfully")
            messages.add_message(request, messages.SUCCESS, message)
```